ballerina/types.bal

```diff
@@ -46,6 +46,69 @@ public type Options record {
     boolean enableConstraintValidation = true;  
     # If `true`, when the result is a list it will contain headers as the first row.
     boolean outputWithHeaders = false;
+    # Specifies the fail-safe options for handling errors during processing
+    FailSafeOptions failSafe?;
+};
+
+# Represents the options for fail-safe mechanism during parsing.
+public type FailSafeOptions record {|
+    # Specifies the output mode for logging errors encountered during parsing
+    ConsoleOutputMode|FileOutputMode outputMode = {};
+|};
+
+# Represents the console output mode for logging errors.
+public type ConsoleOutputMode record {|
+    # Specifies enabling logging errors to the console
+    boolean excludeSourceData = true;
+|};
+
+# Represents the file output mode for logging errors.
+public type FileOutputMode record {|
+    # Specifies enabling logging errors to the console
+    boolean enableConsoleLogs = false;
+    # The file path where errors will be logged
+    string filePath;
+    # Specifies logging source data row along with error details
+    DataType dataType = METADATA;
+    # Configuration for writing to the log file
+    FileWriteOption fileWriteOption = APPEND;
+|};
+
+public enum DataType {
+    # Logs only the error messages without source data rows
+    METADATA,
+    # Logs the raw source data rows along with error messages
+    RAW,
+    # Logs both raw source data rows and metadata along with error messages
+    RAW_AND_METADATA
+};
+
+# Represents an error log entry.
+public type LogOutput record {|
+    # The timestamp of the error occurrence
+    string time;
+    # The location where the error occurred
+    Location location;
+    # The error message
+    string message;
+    # The source data row related to the error
+    string offendingRow;
```

Comment:
Does this represent the raw data line which causes the issue?

Reply:
Yeah